### DISCUSSION TITLE
Allow for new version updates with rebaseWhen=never

### DISCUSSION THREAD

**Discussion by bobvandevijver** [renovatebot/renovate#10564] [Suggest an Idea]
According to [this comment](https://github.com/renovatebot/renovate/issues/10548#issuecomment-866657105) in #10548, the option `rebaseWhen=never` should never update a branch/MR once it has been created by renovate. However, I would like renovate to never rebase the branch when the base branch (master/main) has been updated, but it should update the branch/MR whenever a newer release of the dependency is available.

This is mostly a resource efficiency thing as every commit/branch triggers a pipeline, which can be considered a complete resource waste in certain scenarios. Consider the case where you require the branch to be up to date with the base branch before it can be merged, but you also require 7 days as minimum stability days. You will keep triggering rebases (see the composer note below) when merging other unrelated MRs while it cannot be merged due to the stability days requirement. However, keeping it up to date with the latest released version is desirable, as otherwise you would directly get a new MR after merging the previous one, and you can inspect the previous pipeline result to see which intermediate version broke something.

The `conflicted` option is essentially the same as the `behind-base-branch` one when you're using composer as its lock file contains a content hash.

That why I would like to see either another configuration value between `never` and `conflicted`, something like `on-updated`. It would update the branch (and possibly rebase it) whenever a new version of the dependency is released.

However, as I do not necessarily see this to be fully related to the git related rebase term, an extra option to always force dependency updates regardless of the rebaseWhen setting would be my preferred choice.

**Comment by viceice** [comment]
if you like to have branches created after `stabillityDays` are passed, you can set `internalChecksFilter=strict`, that way renovate supresses branches until `stabillityDays` passes. Then you can switch back `rebaseWhen=auto` so renovate can rebase automerge and conflicted branches.

https://docs.renovatebot.com/configuration-options/#internalchecksfilter

**Reply by bobvandevijver** [reply]
I will try this in combination with the dashboard, as this might actually work better than the stability days jobs that is (still) sometimes registered as an extra pipeline in GitLab (which I tried to solve with #8771).

**Reply by bobvandevijver** [reply]
When setting the `rebaseWhen=auto` I still get too many rebases as every composer merge causes a conflict when its merged. 😕

(below is copied from https://github.com/renovatebot/renovate/discussions/10566, but it adds valuable insight here as well)

We want (and need) to manually verify and possible update the changes made (partly because of #8866), and when approved we can start the process to merge it. The rebase is then either done using the Gitlab rebase button (when possible, which is not the case when another PHP dependency has just been merged), by checking the rebase checkbox or manually (which is significantly faster then waiting for renovate to rebase the branch).

The "problem" is mostly that we have limited processing power to handle all MRs being rebased every half hour, which fully clogs the available runners.

In the past, this workflow work fine, but somehow with a renovate update it broke as it started to rebase everything while it was marked with never (hence #10358). After that fix, the MRs where no longer updated with new version, which it did do before. I think this happened somewhere in v25, as we missed that initial upgrade because renovate didn't upgrade major docker version before.

**Reply by rarkins** [reply]
So you want:
- To ignore conflicted branches, leave them as is
- To update a branch if a newer version exists?

**Reply by bobvandevijver** [reply]
Exactly! 😀

**Reply by rarkins** [reply]
I guess that could be named something like rebaseWhen=newer-version
